### ISSUE TITLE
wrong implementation of text halo color

### ISSUE THREAD

**Issue by sikandernoori** [greensopinion/dart-vector-tile-renderer#81]
https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#paint-symbol-text-halo-color

> The color of the text's halo, which helps it stand out from backgrounds.

But in code it's mapped to `shadow property` of `TextStyle`, while it should be `backgroundColor`

Below is screenshot from Maptiler server serving toner-v2 style: You can the color is mapped to background color of Text not creating shadow.
<img width="1507" alt="Screenshot 2023-04-15 at 12 24 38 PM" src="https://user-images.githubusercontent.com/16399272/232205347-c93ff53e-c7a5-45c4-9ec7-450011d72357.png">

Below is screenshot from dart-vector-tile-renderer plugin, generated using flutter-vector-map-tiles:

<img width="912" alt="Screenshot 2023-04-15 at 12 28 01 PM" src="https://user-images.githubusercontent.com/16399272/232205501-9e338395-8c90-4931-9850-74e7073789f0.png">

**Comment by greensopinion** [comment]
Thanks for opening the issue. There is a problem with the current implementation of text halo that shows at higher zoom levels. The screenshot above demonstrates how the halo is offset too far, letting the background bleed through close to the text.

I disagree with the characterization of the issue as described here. Background color creates a block of color eliminating any detail that would otherwise show through the text. Halo is intended to be a thin area around text that helps text stand out, but still leaves gaps for detail behind the text to show through. 

PR #92 does not correctly address the issue.

**Comment by sikandernoori** [comment]
I will look further into it, but with current shadow implementation label looks like painted more than once.

**Comment by greensopinion** [comment]
Agreed. The current implementation uses four [Shadow](https://api.flutter.dev/flutter/dart-ui/Shadow-class.html)s with an offset to approximate a halo. Likely calculation of the offset is off at some zoom levels.

[text_halo_factory.dart#L38-L59](https://github.com/greensopinion/dart-vector-tile-renderer/blob/d529167940256bf50a5ba7d0ab5c727fb40ba02a/lib/src/themes/text_halo_factory.dart#L38-L59) is a good place to start if you're interested in fixing this issue.

**Comment by JetA2** [comment]
The issue is that the shadow offset is cached once per zoom level, when it depends on the zoom scale factor which changes within a zoom level.
This means the halo expression can't be cached.

Removing the ```cache()``` wrapper at https://github.com/greensopinion/dart-vector-tile-renderer/blob/68966b541302f0d05fbbe7222b966070b897d0c0/lib/src/themes/text_halo_factory.dart#L10 fixes the issue.

This seems a bit overkill for a PR. @greensopinion can you fix at your end when you have time?

**Comment by sikandernoori** [comment]
I am not following this issue anymore. someone else is welcome to take on.
